From cca5a7ce01d35d49ac5750d55c546c0cda24be13 Mon Sep 17 00:00:00 2001
From: ywchen <ywchen@realtek.com>
Date: Wed, 19 Dec 2018 10:56:19 +0800
Subject: [PATCH] add radar for 11h

---
 include/iwinfo.h | 1 +
 iwinfo_lua.c     | 4 ++++
 iwinfo_nl80211.c | 2 ++
 3 files changed, 7 insertions(+)

diff --git a/include/iwinfo.h b/include/iwinfo.h
index 49ee7f0..f1424ef 100644
--- a/include/iwinfo.h
+++ b/include/iwinfo.h
@@ -148,6 +148,7 @@ struct iwinfo_freqlist_entry {
 	uint32_t mhz;
 	uint8_t restricted;
 	uint32_t flags;
+	uint8_t radar;
 };
 
 struct iwinfo_crypto_entry {
diff --git a/iwinfo_lua.c b/iwinfo_lua.c
index eebab8e..e3e851e 100644
--- a/iwinfo_lua.c
+++ b/iwinfo_lua.c
@@ -470,6 +470,10 @@ static int iwinfo_L_freqlist(lua_State *L, int (*func)(const char *, char *, int
 			lua_pushboolean(L, e->restricted);
 			lua_setfield(L, -2, "restricted");
 
+			/* Restricted (Radar) */
+			lua_pushboolean(L, e->radar);
+			lua_setfield(L, -2, "radar");
+
 			lua_rawseti(L, -2, x);
 		}
 	}
diff --git a/iwinfo_nl80211.c b/iwinfo_nl80211.c
index ca78742..98d0c9b 100644
--- a/iwinfo_nl80211.c
+++ b/iwinfo_nl80211.c
@@ -2561,6 +2561,8 @@ static int nl80211_get_freqlist_cb(struct nl_msg *msg, void *arg)
 						!freqs[NL80211_FREQUENCY_ATTR_RADAR]
 					) ? 1 : 0;
 
+					e->radar = freqs[NL80211_FREQUENCY_ATTR_RADAR];
+
 					if (freqs[NL80211_FREQUENCY_ATTR_NO_HT40_MINUS])
 						e->flags |= IWINFO_FREQ_NO_HT40MINUS;
 					if (freqs[NL80211_FREQUENCY_ATTR_NO_HT40_PLUS])
-- 
2.7.4

