diff --git a/common/main.c b/common/main.c
index d4261f4..fe27ca6
--- a/common/main.c
+++ b/common/main.c
@@ -336,10 +336,26 @@ start = get_timer(0);
 #ifdef CONFIG_INSTALL_GPIO_NUM
 		if(!getGPIO(CONFIG_INSTALL_GPIO_NUM)){
 			printf("\nPress Install Button\n");
-			setenv("rescue_cmd", "go r");
+			setenv("rescue_cmd", "go ru");
 			boot_mode = BOOT_RESCUE_MODE;
 			abort = 1; // don't auto boot
 		}
+	if(getenv_ulong("afterupgradeuboot", 16, 0) == 1)
+	{
+		printf("\nupgrade firmware image...\n");
+		setenv("rescue_cmd", "go ru");
+		boot_mode = BOOT_RESCUE_MODE;
+		abort = 1;
+	}
+	if(boot_mode == BOOT_RESCUE_MODE)
+	{
+		run_command("mw 0x98007310 0x08001400", 0);
+		run_command("mw 0x98007314 0x00000000", 0);
+		run_command("mw 0x98007104 0x10000002", 0);
+		setenv("bootup", " ");
+		setenv("upgradeimage", "1");
+		saveenv();
+	}
 #endif
 
 #if defined(CONFIG_BOARD_WD_MONARCH)||defined(CONFIG_BOARD_WD_PELICAN)
diff --git a/common/cmd_boot.c b/common/cmd_boot.c
index 2f613d1..4eb46ff
--- a/common/cmd_boot.c
+++ b/common/cmd_boot.c
@@ -428,6 +428,32 @@ int boot_rescue_from_usb(void)
 		return RTK_PLAT_ERR_READ_RESCUE_IMG;
 	}
 
+	/* Uboot */
+	unsigned long afterupgradeuboot = getenv_ulong("afterupgradeuboot", 16, 0);
+	if (afterupgradeuboot == 0)
+	{
+		if ((filename = getenv("ubootfile")) == NULL) {
+			filename =(char*) CONFIG_RESCUE_FROM_USB_UBOOT;
+		}
+		sprintf(tmpbuf, "fatload usb 0:1 0x1500000 %s", filename);
+		if (run_command(tmpbuf, 0) != 0) {
+			printf("\"%s\" is not exist\n\n", filename);
+		}else{
+			printf("Loading \"%s\" to 0x1500000 is OK.\n\n", filename);
+			run_command("mw 0x98007310 0x28001400", 0);
+			run_command("mw 0x98007314 0", 0);
+			run_command("mw 0x98007104 0x70000000", 0);
+			setenv("afterupgradeuboot", "1");
+			saveenv();
+			if (run_command("go 0x1500000", 0) != 0) {
+				goto loading_failed;
+			}
+		}
+	}else{
+		setenv("afterupgradeuboot", " ");
+		saveenv();
+	}
+
 	/* DTB */
 	if ((filename = getenv("rescue_dtb")) == NULL) {
 		filename =(char*) CONFIG_RESCUE_FROM_USB_DTB;
@@ -516,5 +541,9 @@ int boot_rescue_from_usb(void)
 	ret = rtk_call_booti();
 #endif
 	/* Should not reach here */
+	run_command("mw 0x98007100 0x70000002", 0);
+	run_command("mw 0x98007104 0x70000002", 0);
+	run_command("mw 0x98007310 0x28001400", 0);
+	run_command("mw 0x98007314 0x000a0000", 0);
 
 	return ret;
diff --git a/include/configs/rtd1295_qa_spi_16MB_64.h b/include/configs/rtd1295_qa_spi_16MB_64.h
index 71499c8..d7029fb 100644
--- a/include/configs/rtd1295_qa_spi_16MB_64.h
+++ b/include/configs/rtd1295_qa_spi_16MB_64.h
@@ -116,6 +116,7 @@
 	#define CONFIG_RESCUE_FROM_USB_DTB		"rescue.spi.dtb"
 	#define CONFIG_RESCUE_FROM_USB_ROOTFS		"rescue.root.spi.cpio.gz_pad.img"
 	#define CONFIG_RESCUE_FROM_USB_AUDIO_CORE	"bluecore.audio"
+	#define CONFIG_RESCUE_FROM_USB_UBOOT         	"Uboot-hvlrs.bin"
 #endif /* CONFIG_RESCUE_FROM_USB */
 #define CONFIG_ROOTFS_RESCUE_SIZE	0x100000 // 1 MB
 
diff --git a/include/configs/rtd1295_common.h b/include/configs/rtd1295_common.h
index dca094b..2f7767e 100644
--- a/include/configs/rtd1295_common.h
+++ b/include/configs/rtd1295_common.h
@@ -269,7 +269,7 @@
 /* Port 0, DRD, TYPE_C */
 #define USB_PORT0_GPIO_TYPE "ISOGPIO"
 #define USB_PORT0_GPIO_NUM 1
-#define USB_PORT0_TYPE_C_RD_GPIO_TYPE "ISOGPIO"
+#define USB_PORT0_TYPE_C_RD_GPIO_TYPE "none"
 #define USB_PORT0_TYPE_C_RD_GPIO_NUM 34
 /* Port 1, xhci u2 host */
 #define USB_PORT1_GPIO_TYPE "GPIO"
@@ -316,7 +316,7 @@
 
 /* GPIO */
 #define CONFIG_REALTEK_GPIO
-#define CONFIG_INSTALL_GPIO_NUM    		8
+#define CONFIG_INSTALL_GPIO_NUM                60
 #define CONFIG_HDMITx_HPD_IGPIO_NUM		6
 
 /* I2C */
