diff --git a/arch/arm/cpu/armv8/rtd1295/usb.c b/arch/arm/cpu/armv8/rtd1295/usb.c
index d8e95c99b0..d12aa31e77 100644
--- a/arch/arm/cpu/armv8/rtd1295/usb.c
+++ b/arch/arm/cpu/armv8/rtd1295/usb.c
@@ -283,9 +283,15 @@ void rtk_usb_power_on(void)
 	if ((check & 0x7) != 0x7) {
 		LTRACEF("Realtek-usb: cc1 detect\n");
 		type_c_have_device = 1;
+		writel(readl((volatile u32*) USB_TYPEC_CTRL_CC1_0) |
+			    (0x1 << 29),
+			    (volatile u32*) USB_TYPEC_CTRL_CC1_0);
 	} else if ((check & 0x38) != 0x38) {
 		LTRACEF("Realtek-usb: cc2 detect\n");
 		type_c_have_device = 1;
+		writel(readl((volatile u32*) USB_TYPEC_CTRL_CC1_0) |
+			    (0x1 << 29) | (0x1 << 28) | (0x1 << 27),
+			    (volatile u32*) USB_TYPEC_CTRL_CC1_0);
 	}
 
 	//Type C 5V
