diff --git a/common/main.c b/common/main.c
old mode 100644
new mode 100755
index d4261f4..fe27ca6
--- a/common/main.c
+++ b/common/main.c
@@ -336,10 +336,17 @@ start = get_timer(0);
 #ifdef CONFIG_INSTALL_GPIO_NUM
 		if(!getGPIO(CONFIG_INSTALL_GPIO_NUM)){
 			printf("\nPress Install Button\n");
-			setenv("rescue_cmd", "go r");
+			setenv("rescue_cmd", "go ru");
 			boot_mode = BOOT_RESCUE_MODE;
 			abort = 1; // don't auto boot
 		}
+    if(getenv_ulong("afterupgradeuboot", 16, 0) == 1)
+    {
+			printf("\nupgrade firmware image...\n");
+			setenv("rescue_cmd", "go ru");
+      boot_mode = BOOT_RESCUE_MODE;
+      abort = 1;
+    }
 #endif
 
 #if defined(CONFIG_BOARD_WD_MONARCH)||defined(CONFIG_BOARD_WD_PELICAN)
diff --git a/common/cmd_boot.c b/common/cmd_boot.c
old mode 100644
new mode 100755
index 2f613d1..4eb46ff
--- a/common/cmd_boot.c
+++ b/common/cmd_boot.c
@@ -428,6 +428,30 @@ int boot_rescue_from_usb(void)
 		return RTK_PLAT_ERR_READ_RESCUE_IMG;
 	}
 
+  /* Uboot */
+  unsigned long afterupgradeuboot = getenv_ulong("afterupgradeuboot", 16, 0);
+  if (afterupgradeuboot == 0)
+  {
+    if ((filename = getenv("ubootfile")) == NULL) {
+      filename =(char*) CONFIG_RESCUE_FROM_USB_UBOOT;
+    }
+    sprintf(tmpbuf, "fatload usb 0:1 0x1500000 %s", filename);
+    if (run_command(tmpbuf, 0) != 0) {
+      printf("\"%s\" is not exist\n\n", filename);
+    }else{
+      printf("Loading \"%s\" to 0x1500000 is OK.\n\n", filename);
+      run_command("mw 0x98007314 0", 0);
+      setenv("afterupgradeuboot", "1");
+      saveenv();
+      if (run_command("go 0x1500000", 0) != 0) {
+        goto loading_failed;
+      }
+    }
+  }else{
+    setenv("afterupgradeuboot", " ");
+    saveenv();
+  }
+
 	/* DTB */
 	if ((filename = getenv("rescue_dtb")) == NULL) {
 		filename =(char*) CONFIG_RESCUE_FROM_USB_DTB;
diff --git a/include/configs/rtd1295_qa_spi_16MB_64.h b/include/configs/rtd1295_qa_spi_16MB_64.h
index 71499c8..d7029fb 100644
--- a/include/configs/rtd1295_qa_spi_16MB_64.h
+++ b/include/configs/rtd1295_qa_spi_16MB_64.h
@@ -116,6 +116,7 @@
 	#define CONFIG_RESCUE_FROM_USB_DTB		"rescue.spi.dtb"
 	#define CONFIG_RESCUE_FROM_USB_ROOTFS		"rescue.root.spi.cpio.gz_pad.img"
 	#define CONFIG_RESCUE_FROM_USB_AUDIO_CORE	"bluecore.audio"
+	#define CONFIG_RESCUE_FROM_USB_UBOOT         	"Uboot-hvlrs.bin"
 #endif /* CONFIG_RESCUE_FROM_USB */
 #define CONFIG_ROOTFS_RESCUE_SIZE	0x100000 // 1 MB
 
