From b25a4c4d1b94de6f6331c8702d7e2ef8fb3fc2a2 Mon Sep 17 00:00:00 2001
From: "william.lee" <william.lee@realtek.com>
Date: Wed, 6 Mar 2019 16:08:33 +0800
Subject: [PATCH] Multithread Raid5 Driver

---
 drivers/md/raid5.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/md/raid5.c b/drivers/md/raid5.c
index 0841d8f..408346e 100644
--- a/drivers/md/raid5.c
+++ b/drivers/md/raid5.c
@@ -6536,7 +6536,7 @@ static struct r5conf *setup_conf(struct mddev *mddev)
 	if (conf == NULL)
 		goto abort;
 	/* Don't enable multi-threading by default*/
-	if (!alloc_thread_groups(conf, 0, &group_cnt, &worker_cnt_per_group,
+	if (!alloc_thread_groups(conf, NR_CPUS, &group_cnt, &worker_cnt_per_group,
 				 &new_group)) {
 		conf->group_cnt = group_cnt;
 		conf->worker_cnt_per_group = worker_cnt_per_group;
-- 
2.7.4

