diff --git a/drivers/leds/led-core.c b/drivers/leds/led-core.c
index 454ed4d..c5e44c0 100644
--- a/drivers/leds/led-core.c
+++ b/drivers/leds/led-core.c
@@ -168,7 +168,7 @@ static void led_blink_setup(struct led_classdev *led_cdev,
 
 	/* blink with 1 Hz as default if nothing specified */
 	if (!*delay_on && !*delay_off)
-		*delay_on = *delay_off = 500;
+               *delay_on = *delay_off = 200;
 
 	led_set_software_blink(led_cdev, *delay_on, *delay_off);
 }
