Index: backports-2017-11-01/drivers/net/wireless/realtek/Kconfig
===================================================================
--- backports-2017-11-01.orig/drivers/net/wireless/realtek/Kconfig
+++ backports-2017-11-01/drivers/net/wireless/realtek/Kconfig
@@ -14,5 +14,6 @@ if WLAN_VENDOR_REALTEK
 source "drivers/net/wireless/realtek/rtl818x/Kconfig"
 source "drivers/net/wireless/realtek/rtlwifi/Kconfig"
 source "drivers/net/wireless/realtek/rtl8xxxu/Kconfig"
+source "drivers/net/wireless/realtek/rtkwifiu/Kconfig"
 
 endif # WLAN_VENDOR_REALTEK
Index: backports-2017-11-01/drivers/net/wireless/realtek/Makefile
===================================================================
--- backports-2017-11-01.orig/drivers/net/wireless/realtek/Makefile
+++ backports-2017-11-01/drivers/net/wireless/realtek/Makefile
@@ -6,4 +6,5 @@ obj-$(CPTCFG_RTL8180)		+= rtl818x/
 obj-$(CPTCFG_RTL8187)		+= rtl818x/
 obj-$(CPTCFG_RTLWIFI)		+= rtlwifi/
 obj-$(CPTCFG_RTL8XXXU)		+= rtl8xxxu/
+obj-$(CPTCFG_RTKWIFIU)		+= rtkwifiu/
 
Index: backports-2017-11-01/drivers/net/wireless/realtek/rtkwifiu/rtl8811au/Kconfig
===================================================================
--- backports-2017-11-01.orig/drivers/net/wireless/realtek/rtkwifiu/rtl8811au/Kconfig
+++ backports-2017-11-01/drivers/net/wireless/realtek/rtkwifiu/rtl8811au/Kconfig
@@ -1,6 +1,6 @@
-config RTL8821AU
-	tristate "Realtek 8821A USB WiFi"
+config RTL8811AU
+	tristate "Realtek 8811A USB WiFi"
 	depends on USB
 	---help---
-	  Help message of RTL8821AU
+	  Help message of RTL8811AU
 
Index: backports-2017-11-01/drivers/net/wireless/realtek/rtkwifiu/rtl8811au/Makefile
===================================================================
--- backports-2017-11-01.orig/drivers/net/wireless/realtek/rtkwifiu/rtl8811au/Makefile
+++ backports-2017-11-01/drivers/net/wireless/realtek/rtkwifiu/rtl8811au/Makefile
@@ -2113,11 +2113,11 @@ ifeq ($(CONFIG_RTL8723B), y)
 $(MODULE_NAME)-$(CONFIG_MP_INCLUDED)+= core/rtw_bt_mp.o
 endif
 
-obj-$(CONFIG_RTL8821AU) := $(MODULE_NAME).o
+obj-$(CONFIG_RTL8811AU) := $(MODULE_NAME).o
 
 else
 
-export CONFIG_RTL8821AU = m
+export CONFIG_RTL8811AU = m
 
 all: modules
 
Index: backports-2017-11-01/drivers/net/wireless/realtek/rtkwifiu/rtl8812bu/Kconfig
===================================================================
--- backports-2017-11-01.orig/drivers/net/wireless/realtek/rtkwifiu/rtl8812bu/Kconfig
+++ backports-2017-11-01/drivers/net/wireless/realtek/rtkwifiu/rtl8812bu/Kconfig
@@ -1,6 +1,6 @@
-config RTL8822BU
-	tristate "Realtek 8822B USB WiFi"
+config RTL8812BU
+	tristate "Realtek 8812B USB WiFi"
 	depends on USB
 	---help---
-	  Help message of RTL8822BU
+	  Help message of RTL8812BU
 
Index: backports-2017-11-01/drivers/net/wireless/realtek/rtkwifiu/rtl8812bu/Makefile
===================================================================
--- backports-2017-11-01.orig/drivers/net/wireless/realtek/rtkwifiu/rtl8812bu/Makefile
+++ backports-2017-11-01/drivers/net/wireless/realtek/rtkwifiu/rtl8812bu/Makefile
@@ -1930,11 +1930,11 @@ ifeq ($(CONFIG_RTL8723B), y)
 $(MODULE_NAME)-$(CONFIG_MP_INCLUDED)+= core/rtw_bt_mp.o
 endif
 
-obj-$(CONFIG_RTL8822BU) := $(MODULE_NAME).o
+obj-$(CONFIG_RTL8812BU) := $(MODULE_NAME).o
 
 else
 
-export CONFIG_RTL8822BU = m
+export CONFIG_RTL8812BU = m
 
 all: modules
 
Index: backports-2017-11-01/drivers/net/wireless/realtek/rtkwifiu/rtl8822cs/Kconfig
===================================================================
--- backports-2017-11-01.orig/drivers/net/wireless/realtek/rtkwifiu/rtl8822cs/Kconfig
+++ backports-2017-11-01/drivers/net/wireless/realtek/rtkwifiu/rtl8822cs/Kconfig
@@ -1,4 +1,4 @@
-config RTL8822BS
+config RTL8822CS
 	tristate "Realtek 8822C SDIO WiFi"
 	---help---
 	  Help message of RTL8822CS
Index: backports-2017-11-01/drivers/net/wireless/realtek/rtkwifiu/rtl8822cu/Kconfig
===================================================================
--- backports-2017-11-01.orig/drivers/net/wireless/realtek/rtkwifiu/rtl8822cu/Kconfig
+++ backports-2017-11-01/drivers/net/wireless/realtek/rtkwifiu/rtl8822cu/Kconfig
@@ -1,4 +1,4 @@
-config RTL8822BU
+config RTL8822CU
 	tristate "Realtek 8822C USB WiFi"
 	depends on USB
 	---help---
