diff --git a/src/lib/compress.c b/src/lib/compress.c
index bba4fcf..e1fd3a2 100644
--- a/src/lib/compress.c
+++ b/src/lib/compress.c
@@ -81,8 +81,8 @@
 
 #define COMPR_ERR_MAX 128
 
-/* Default maximum time we will wait in a poll() - 20 seconds */
-#define DEFAULT_MAX_POLL_WAIT_MS    20000
+/* Default maximum time we will wait in a poll() - 10 milliseconds */
+#define DEFAULT_MAX_POLL_WAIT_MS    10
 
 struct compress {
 	int fd;
