diff --git a/src/utils/cplay.c b/src/utils/cplay.c
index 3e9fcac..0f23551 100644
--- a/src/utils/cplay.c
+++ b/src/utils/cplay.c
@@ -307,6 +307,7 @@ void play_samples(char *name, unsigned int card, unsigned int device,
 	FILE *file;
 	char *buffer;
 	int size, num_read, wrote;
+	int left, written;
 
 	if (verbose)
 		printf("%s: entry\n", __func__);
@@ -381,28 +382,32 @@ void play_samples(char *name, unsigned int card, unsigned int device,
 
 	do {
 		num_read = fread(buffer, 1, size, file);
-		if (num_read > 0) {
-			wrote = compress_write(compress, buffer, num_read);
+		left = num_read;
+		written = 0;
+		while (left > 0) {
+			wrote = compress_write(compress, buffer+written, left);
 			if (wrote < 0) {
 				fprintf(stderr, "Error playing sample\n");
 				fprintf(stderr, "ERR: %s\n", compress_get_error(compress));
 				goto BUF_EXIT;
 			}
-			if (wrote != num_read) {
+			if (wrote != left) {
 				/* TODO: Buffer pointer needs to be set here */
-				fprintf(stderr, "We wrote %d, DSP accepted %d\n", num_read, wrote);
+				fprintf(stderr, "We wrote %d, DSP accepted %d\n", left, wrote);
 			}
 			if (verbose) {
 				print_time(compress);
 				printf("%s: wrote %d\n", __func__, wrote);
 			}
+			left -= wrote;
+			written += wrote;
 		}
 	} while (num_read > 0);
 
 	if (verbose)
 		printf("%s: exit success\n", __func__);
 	/* issue drain if it supports */
-	compress_drain(compress);
+	compress_stop(compress);
 	free(buffer);
 	fclose(file);
 	compress_close(compress);
